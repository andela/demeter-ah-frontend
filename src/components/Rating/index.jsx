import React, { Fragment } from 'react';
import Star from '../../assets/svgs/starblank';
import Starcolor from '../../assets/svgs/star';

const Stars = ({ rating }) => {
  const loadStar = () => {
    const rate = rating ? Math.floor(rating) : 0;
    const stars = [];
    for (let i = 0; i < 5; i += 1) {
      if (i < rate) {
        stars.push(<Starcolor key={i} />);
      } else {
        stars.push(<Star key={i} />);
      }
    }
    return stars;
  };
  return (
    loadStar()
  );
};

export default Stars;
