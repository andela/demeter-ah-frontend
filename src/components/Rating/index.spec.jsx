import React from 'react';
import { shallow } from 'enzyme';
import Rating from './index';

describe('Rating Component', () => {
  it('Should render without errors', () => {
    const component = shallow(<Rating />);
    expect(component).toMatchSnapshot();
  });
});
