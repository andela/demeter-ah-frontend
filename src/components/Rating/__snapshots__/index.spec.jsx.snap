// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating Component Should render without errors 1`] = `
Array [
  <Star
    key="0"
  />,
  <Star
    key="1"
  />,
  <Star
    key="2"
  />,
  <Star
    key="3"
  />,
  <Star
    key="4"
  />,
]
`;
